Seminars/Sem.02/Pract.02/Task07.cpp

```diff
@@ -46,55 +46,49 @@ void inputUsers(System &newSystem)
 
 void printUsersList(System &newSystem)
 {
-    std::ifstream readFromFile(FILENAME1);
+    std::ifstream readFromFile(FILENAME);
     if (!readFromFile.is_open())
     {
         std::cout << "ERROR! The file could not be opened!";
         return;
     }
 
-    int i;
-    for (i = 0; i < USERS; ++i)
+    int count = 0;
+    for (int i = 0; i < USERS; ++i)
     {
         readFromFile >> newSystem.systemOfUsers[i].name
             >> newSystem.systemOfUsers[i].email
             >> newSystem.systemOfUsers[i].password;
+
+        ++count;
     }
-    newSystem.countOfStudentsInSystem = i;
+    newSystem.countOfStudentsInSystem = count;
 
     readFromFile.close();
 }
 
 void makeRegistration(System& newSystem)
 {
-    int i = 0;
     std::cout << "Enter your name: ";
-    std::cin >> newSystem.systemOfUsers[i].name;
-    ++i;
+    std::cin >> newSystem.systemOfUsers[0].name;
 
     std::cout << "Enter your email: ";
-    std::cin >> newSystem.systemOfUsers[i].email;
-    ++i;
+    std::cin >> newSystem.systemOfUsers[1].email;
```

Comment:
Why do you store the email of the new user at index 1, which is different index from index 0